subxt/src/config/signed_extensions.rs

```diff
@@ -213,13 +223,16 @@ impl<T: Config> ExtrinsicParamsEncoder for CheckMortality<T> {
 
 impl<T: Config> SignedExtension<T> for CheckMortality<T> {
     const NAME: &'static str = "CheckMortality";
+    type Decoded = Era;
 }
 
 /// The [`ChargeAssetTxPayment`] signed extension.
-#[derive(Debug)]
+#[derive(Debug, Encode, DecodeAsType)]
 pub struct ChargeAssetTxPayment {
-    tip: Compact<u128>,
-    asset_id: Option<u32>,
+    /// Tip
+    pub tip: Compact<u128>,
+    /// Asset Id
+    pub asset_id: Option<u32>,
```

Comment:
here and below I think I wouldn't make these Pub and instead expose `tip()` and `asset_id()` getters where `tip` returns `u128` and the `Compact` thing is hidden away (it's an implementation detail) :)